include/alpaka/mem/buf/cpu/Copy.hpp

```diff
@@ -177,6 +176,81 @@ namespace alpaka
+
+            static_assert(
+                Dim<TViewDst>::value == Dim<TViewSrc>::value,
+                "The source and the destination view are required to have the same dimensionality!");
+            static_assert(
+                Dim<TViewDst>::value == Dim<TExtent>::value,
+                "The views and the extent are required to have the same dimensionality!");
+            static_assert(
+                Dim<TViewDst>::value == 0u,
+                "The destination view and the input TDim are required to have the same dimensionality!");
+
+            static_assert(
+                meta::IsIntegralSuperset<DstSize, ExtentSize>::value,
+                "The destination view and the extent are required to have compatible idx type!");
+            static_assert(
+                meta::IsIntegralSuperset<SrcSize, ExtentSize>::value,
+                "The source view and the extent are required to have compatible idx type!");
+
+            static_assert(
+                std::is_same<alpaka::Elem<TViewDst>, std::remove_const_t<alpaka::Elem<TViewSrc>>>::value,
+                "The source and the destination view are required to have the same element type!");
+
+            TaskCopyCpu(TViewDst& viewDst, TViewSrc const& viewSrc, TExtent const& extent)
+                : m_dstMemNative(reinterpret_cast<std::uint8_t*>(getPtrNative(viewDst)))
+                , m_srcMemNative(reinterpret_cast<std::uint8_t const*>(getPtrNative(viewSrc)))
+            {
+                // all zero-sized extents are equivalent
+                ALPAKA_ASSERT(extent::getExtentVec(extent).prod() == 1u);
+                ALPAKA_ASSERT(extent::getExtentVec(viewDst).prod() == 1u);
+                ALPAKA_ASSERT(extent::getExtentVec(viewSrc).prod() == 1u);
+#ifdef NDEBUG
+                alpaka::ignore_unused(extent);
+#endif
+            }
+
+#if ALPAKA_DEBUG >= ALPAKA_DEBUG_FULL
+            ALPAKA_FN_HOST auto printDebug() const -> void
+            {
+                std::cout << __func__ << " e: " << Scalar() << " ewb: " << sizeof(Elem) << " de: " << Scalar()
+                          << " dptr: " << reinterpret_cast<void*>(m_dstMemNative) << " dpitchb: " << Scalar()
+                          << " se: " << Scalar() << " sptr: " << reinterpret_cast<void const*>(m_srcMemNative)
+                          << " spitchb: " << Scalar() << std::endl;
+            }
+#endif
+
+            ALPAKA_FN_HOST auto operator()() const -> void
+            {
+                ALPAKA_DEBUG_MINIMAL_LOG_SCOPE;
+
+#if ALPAKA_DEBUG >= ALPAKA_DEBUG_FULL
+                printDebug();
+#endif
+                std::memcpy(
+                    reinterpret_cast<void*>(m_dstMemNative),
+                    reinterpret_cast<void const*>(m_srcMemNative),
+                    sizeof(Elem));
+            }
+
+            std::uint8_t* const m_dstMemNative;
+            std::uint8_t const* const m_srcMemNative;
```

Comment:
I would prefer `std::byte` here, but I guess alpaka uses `std::uint8_t` in many other places as well. I opened #1545 to refactor this at some point.

Reply:
I've had mixed experience with `std::byte`, but yes, I agree it would be the type whose intent best matches the use case.
However, I have not seen it used anywhere in Alpaka so far.